eng/pipelines/checkout-windows-task.yml

```diff
@@ -0,0 +1,11 @@
+# Shallow checkout sources on Windows
+steps:
+  - checkout: none
+
+  - script: |
+      @echo on
+      git init
+      git remote add origin "$(Build.Repository.Uri)"
+      git fetch --progress --no-tags --depth=1 origin "$(Build.SourceVersion)"
```

Comment:
One of the reasons we used `depth=5` is because there is a bit of a race condition in the old model with shallow clones. By fetching the `$(Build.SourceVersion)` are we just side stepping that problem.

Reply:
Oops, I didn't notice that this was a question, and not a statement.

I think we are. The `Build.SourceVersion` is a commit SHA which is fixed for a given run of a pipeline. You can see its value on the [pipeline run page](https://dev.azure.com/dnceng/public/_build/results?buildId=922936&view=results) next to the PR number.

Reply:
@RikkiGibson 

Did you validate that `$(Build.SourceVersion)` represents the result of merging the branch back into the target branch vs. being the HEAD of the branch being merged? The docs are ambiguous as to which this is.

Reply:
Yes, when you view the commit details in the pipeline run page, you will find it's always a merge commit with the message "Merge {a} into {b}", where one of its parents is the PR's head branch and the other parent is the base branch.